proto/message_contents/signature.proto

```diff
@@ -22,8 +22,16 @@ message Signature {
         bytes bytes = 1; // compact representation [ R || S ], 64 bytes
         uint32 recovery = 2; // recovery bit
     }
+    // Self-describing signature format meant for maximizing interoperability
+    message Generic {
+        bytes signature_bytes = 1; // signature bytes
+        bytes message_bytes = 2;
```

Comment:
In what cases would the `message_bytes` be full content and in what cases would it be a digest?

For V2 messages, we use currently sign the digest of `concat(headerBytes, messageContents)`. If the expectation was to use the raw inputs instead of the digest, we would effectively be doubling the size of our messages.

Reply:
Great point. 

Seems like Association Signature, and DigestSignature should be handled differently. From the lens of 'self describing' a signature to prove integrity needs the hashing algorithm and approach (eg. EIP 191)

Where Signatures used to prove user approval need extra context such as the text that was submitted. 

Toy Example:
```
AssociationProof :{
    Message,
    Signature: {
          type: ECDSA_keccak256_eip191,
          bytes:  ...
    }
}
```

Reply:
Is the proposal to create a new proto at a higher level for "proof"? That could bundle the context and the signature, separating signature math specifics from the use case and message verification.